cli/root.go

```diff
@@ -139,5 +139,29 @@ Start a DefraDB node, interact with a local or remote node, and much more.
 		"Path to the private key for tls",
 	)
 
+	cmd.PersistentFlags().String(
+		"keyring-namespace",
+		"defradb",
+		"Service name to use when using the system backend",
+	)
+
+	cmd.PersistentFlags().String(
+		"keyring-backend",
+		"file",
+		"Keyring backend to use. Options are file or system",
+	)
+
+	cmd.PersistentFlags().String(
+		"keyring-path",
+		"keys",
+		"Path to store encrypted keys when using the file backend",
+	)
+
+	cmd.PersistentFlags().Bool(
+		"no-keyring",
+		false,
+		"Disable the keyring and generate ephemeral keys",
```

Comment:
question: Would this disable encryption at rest? If so, I suggest noting that in this description.

question: What happens if keyring commands are called if the keyring is disabled?

question: Is this flag available for all commands?  Not just `start`/`init`?  If so what happens if you disable it whilst/after it has been in use?

Reply:
This is what happens when you use `--no-keyring` after it has been in use:
```
defradb start --no-keyring
2024-05-07T14:44:30.085-0400	ERROR	badger	v4@v4.2.1-0.20231113215945-a63444ca5276/db.go:263	Received err: Encryption key mismatch. Cleaning up...
Error: Encryption key mismatch
```

Reply:
> question: What happens if keyring commands are called if the keyring is disabled?

`--no-keyring` is unused for keyring commands. It won't affect the outcome.

Reply:
Calling it `start` is of less a worry to me than if someone uses it in an active instance, for example:
```
defradb client collection get --name User bae-123 --no-keyring
// or
defradb client collection create --name User '{ "name": "Bob" }'  --no-keyring
```

Reply:
> question: Would this disable encryption at rest? If so, I suggest noting that in this description.

Yes, I'll update the description.

> question: What happens if keyring commands are called if the keyring is disabled?

You can still use the keyring commands to manage keys. 

> question: Is this flag available for all commands? Not just start/init? If so what happens if you disable it whilst/after it has been in use?

This currently only applies to the start command. I'm planning on moving the flags that only apply to the start command in a future PR. 

If you disable the keyring after using it, the peer key will be randomly generated, and at rest encryption will be disabled. 

If at rest encryption was previously enabled, the datastore will fail to open due to the encryption key not matching.

Reply:
> --no-keyring is unused for keyring commands. It won't affect the outcome.

It is a no-op?  Might it be worth excluding it as a flag from those commands then?

Reply:
> This currently only applies to the start command. I'm planning on moving the flags that only apply to the start command in a future PR.

Okay nice, I'm happy with that - thanks Keenan :)